components/telemetry-operator/internal/validation/plugin_validator.go

```diff
@@ -4,102 +4,119 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/pkg/errors"
+
 	telemetryv1alpha1 "github.com/kyma-project/kyma/components/telemetry-operator/api/v1alpha1"
 )
 
 //go:generate mockery --name PluginValidator --filename plugin_validator.go
 type PluginValidator interface {
 	Validate(logPipeline *telemetryv1alpha1.LogPipeline, logPipelines *telemetryv1alpha1.LogPipelineList) error
+	ContainsCustomPlugin(logPipeline *telemetryv1alpha1.LogPipeline) bool
 }
 
 type pluginValidator struct {
-	supportedFilterPlugins []string
-	supportedOutputPlugins []string
-	deniedFilterPlugins    []string
-	deniedOutputPlugins    []string
+	deniedFilterPlugins []string
+	deniedOutputPlugins []string
 }
 
-func NewPluginValidator(supportedFilerPlugins, supportedOutputPlugins, deniedFilterPlugins, deniedOutputPlugins []string) PluginValidator {
+func NewPluginValidator(deniedFilterPlugins, deniedOutputPlugins []string) PluginValidator {
 	return &pluginValidator{
-		supportedFilterPlugins: supportedFilerPlugins,
-		supportedOutputPlugins: supportedOutputPlugins,
-		deniedFilterPlugins:    deniedFilterPlugins,
-		deniedOutputPlugins:    deniedOutputPlugins,
+		deniedFilterPlugins: deniedFilterPlugins,
+		deniedOutputPlugins: deniedOutputPlugins,
 	}
 }
 
-func (v *pluginValidator) Validate(logPipeline *telemetryv1alpha1.LogPipeline, logPipelines *telemetryv1alpha1.LogPipelineList) error {
-
+// ContainsCustomPlugin returns true if the pipeline
+// contains any custom filters or outputs
+func (pv *pluginValidator) ContainsCustomPlugin(logPipeline *telemetryv1alpha1.LogPipeline) bool {
 	for _, filterPlugin := range logPipeline.Spec.Filters {
-		if err := checkIfPluginIsValid("filter", filterPlugin.Content, logPipeline, v.supportedFilterPlugins, v.deniedFilterPlugins, logPipelines); err != nil {
-			return err
+		if filterPlugin.Custom != "" {
+			return true
+		}
+	}
+	if logPipeline.Spec.Output.Custom != "" {
+		return true
+	}
+	for _, f := range logPipeline.Spec.Filters {
+		if f.Custom != "" {
+			return true
 		}
 	}
+	return false
+}
 
-	for _, outputPlugin := range logPipeline.Spec.Outputs {
-		if err := checkIfPluginIsValid("output", outputPlugin.Content, logPipeline, v.supportedOutputPlugins, v.deniedOutputPlugins, logPipelines); err != nil {
+// Validate returns an error if validation fails
+// because of using denied plugins or errors in match conditions
+// for filters or outputs.
+func (pv *pluginValidator) Validate(logPipeline *telemetryv1alpha1.LogPipeline, logPipelines *telemetryv1alpha1.LogPipelineList) error {
+	err := pv.validateFilters(logPipeline, logPipelines)
+	if err != nil {
+		return errors.Wrap(err, " error validating Filter plugins")
```

Comment:
```suggestion
		return errors.Wrap(err, " error validating filter plugins")
```

I assume there's a reason why the sentence starts with a blank space and lower case :)